src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1336,7 +1424,21 @@ protected override void ProcessRecord()
                                     contentType);
                                 WriteVerbose(respVerboseMsg);
 
-                                if (!response.IsSuccessStatusCode)
+                                bool _isSuccess = response.IsSuccessStatusCode;
+
+                                // Check if the Resume range was not satisfiable because the file already completed downloading.
+                                // This happens when the local file is the same size as the remote file.
+                                if (Resume.IsPresent &&
+                                    response.StatusCode == HttpStatusCode.RequestedRangeNotSatisfiable && 
+                                    response.Content.Headers.ContentRange.HasLength && 
+                                    response.Content.Headers.ContentRange.Length == _resumeFileSize)
+                                {
+                                    _isSuccess = true;
```

Comment:
Should we have a Verbose message informing the user nothing was downloaded?  Edge case where the local file is same size as remote file, but the files are different.

Reply:
good idea. I'll add it.

Reply:
suggestions for wording?

> The remote server indicates the file is already complete. OutFile will not be written to.

Reply:
How about:

> The OutFile "{0}" is the same size as the remote file.  The file will not be re-downloaded.

Reply:
Much better. fixed

Reply:
The file name can be long. I suggest move it to end:
` The file will not be re-downloaded because the remote file is the same size as the OutFile: {0}`.

Reply:
agreed. fixed